gobblin-data-management/src/main/java/gobblin/data/management/copy/RecursiveCopyableDataset.java

```diff
@@ -33,31 +34,41 @@
  * Implementation of {@link CopyableDataset} that creates a {@link CopyableFile} for every file that is a descendant if
  * the root directory.
  */
-public class RecursiveCopyableDataset implements CopyableDataset {
+public class RecursiveCopyableDataset implements CopyableDataset, FileSystemDataset {
 
   private final Path rootPath;
   private final FileSystem fs;
   private final PathFilter pathFilter;
+  // Glob used to find this dataset
+  private final Path glob;
   private final CopyableFileFilter copyableFileFilter;
 
-  public RecursiveCopyableDataset(final FileSystem fs, Path rootPath, Properties properties) {
+  public RecursiveCopyableDataset(final FileSystem fs, Path rootPath, Properties properties, Path glob) {
 
     this.rootPath = PathUtils.getPathWithoutSchemeAndAuthority(rootPath);
     this.fs = fs;
 
     this.pathFilter = DatasetUtils.instantiatePathFilter(properties);
     this.copyableFileFilter = DatasetUtils.instantiateCopyableFileFilter(properties);
+    this.glob = glob;
   }
 
   @Override public Collection<CopyableFile> getCopyableFiles(FileSystem targetFs, CopyConfiguration configuration)
       throws IOException {
 
+    Path nonGlobSearchPath = PathUtils.deepestNonGlobPath(this.glob);
+
     List<FileStatus> files = FileListUtils.listFilesRecursively(this.fs, this.rootPath, this.pathFilter);
 
     List<CopyableFile> copyableFiles = Lists.newArrayList();
 
     for (FileStatus file : files) {
-      copyableFiles.add(CopyableFile.builder(this.fs, file, this.rootPath, configuration).fileSet(file.getPath().getParent().toString()).build());
+      Path filePathRelativeToSearchPath = PathUtils.relativizePath(PathUtils.getPathWithoutSchemeAndAuthority(file.getPath()),
```

Comment:
I changed the semantics of `PathUtils.relativizePath` instead (in reality, we've always assumed the new semantics).